### ISSUE TITLE
Axes break when a label contains an unknown char

### ISSUE THREAD

**Issue by flekschas** [wwwtyro/candygraph#28]
Say you create your most favorite a-axis. Say we take the x-axis from https://github.com/wwwtyro/candygraph/blob/master/docs/examples/src/ex-00100-simple-plot.ts#L53-L62 and add a magical character to the `labelFormatter` as follows:

```js
createOrthoAxis(cg, coords, "x", font, {
      labelSide: 1,
      tickOffset: -2.5 * dpr,
      tickLength: 6 * dpr,
      tickStep: 0.2,
      tickWidth: 1 * dpr,
      axisWidth: 1 * dpr,
      labelSize: 12 * dpr,
      labelFormatter: (n) => `${String.fromCharCode(8722)}${n.toFixed(1)}`,
    })
```

Then CadyGraph will fail rendering due to the following error thrown by https://github.com/wwwtyro/candygraph/blob/master/src/primitives/text.ts#L131:

```
Cannot read properties of undefined (reading 'xoffset')
```

A simple fix might be to extend the branching condition in https://github.com/wwwtyro/candygraph/blob/master/src/primitives/text.ts#L125 to

```js
        // Find the glyph for that character.
        const glyph = this.font.glyphs[char.charCodeAt(0)];

        // <NEW>
        // Unknown glyph. Let's continue without it.
        if (!glyph) continue;
        // </NEW>

        // If this isn't a space character, go ahead and append layout data.
        if (char !== " ") {
```

In my quick test it works but I want to check in first :) Maybe there's a better approach at fixing this. We might also want to think of logging a warning so the user isn't confused why a char might be missing.

tl/dr: In case you wonder what `String.fromCharCode(8722)`: it's "−". I came across this issue as d3.format uses that char for negative values and using d3.format with `labelFormatter` lead to the issue I described earlier. This makes me wonder if there'd be a benefit from extending the default font with that one char just because D3 is so popular and other folks might use it for formatting as well.

**Comment by wwwtyro** [comment]
Thanks @flekschas! What do you think about replacing any unknown glyphs with the [empty square glyph](https://stackoverflow.com/questions/13730544/is-there-a-glyph-not-found-character): □ ?

Logging a warning sounds good to me, and we might want to cut it off after some number of them (maybe 100?) to prevent possible perf hits.

I think adding requested chars to the default font is a great idea and "−" seems like a good candidate. Maybe someday we have to worry about the size of the default, but people can always build their own, or we can offer multiple prebuilt options.

**Comment by flekschas** [comment]
Cool, I'll take a stab at it.

Regarding the empty square glyph, I like the idea but I am not sure there's a bullet proof way to display it without hard-coding the glyph into the `Text` class as we cannot guarantee that whatever font is passed to `Text` actually contains the empty square glyph. The default font, for instance, does not contain it.

I think a safer way is to just skip the character. What do you think? Also, the problem might fix itself if we move to https://github.com/mapbox/tiny-sdf as suggested by #29.
